### PR TITLE
[build-script] Add a flag for sccache

### PR DESCRIPTION
**Pull request by porglezomp**
This adds a flag to enable sccache in order to simplify the build-script invocation, particularly for new contributors.

The implementation is based on the implementation of the `--distcc` flag.
CC @gottesmm

### CODE REVIEW

**Review by shahmishal — APPROVED** [review]
Thanks for doing this!

**Comment by shahmishal** [comment]
@swift-ci Python lint

**Comment by gottesmm** [comment]
Thanks!

**Review by gottesmm — APPROVED** [review]

**Comment by shahmishal** [comment]
@swift-ci Python lint

**Comment by shahmishal** [comment]
@swift-ci smoke test

File: docs/DevelopmentTips.md [hunk 1]
```diff
@@ -33,9 +33,11 @@ Compilation times for the compiler and the standard library can be agonizing, es
 ```
 $ brew install sccache
 $ sccache --start-server
-$ ./swift/utils/build-script MY_ARGS --cmake-c-launcher $(which sccache) --cmake-cxx-launcher $(which sccache)
+$ ./swift/utils/build-script MY_ARGS --sccache
 ```
 
+If you want to always use sccache, you can `export USE_SCCACHE=1` and the build script will pick it up.
```
Comment (kastiglione) [inline docs/DevelopmentTips.md:39]: should env vars be prefixed? `SWIFT_USE_SCCACHE`?

Comment (porglezomp) [reply docs/DevelopmentTips.md:39]: I was copying the naming scheme used by the environment variable `distcc`, but I wouldn't be opposed to prefixing it.

Comment (kastiglione) [reply docs/DevelopmentTips.md:39]: Yeah, I figured it was as good of timing as any to ask.

**Comment by varungandhi-apple** [comment]
Maybe I'm missing something in the PR, but what part of the code is actually finding the path to `sccache` and setting `toolchain.sccache` to that path?

**Comment by porglezomp** [comment]
The toolchains do a generic lazy lookup for tools, so this just adds this line that makes it eligible:
```python
_register("sccache", "sccache")
```

**Comment by shahmishal** [comment]
@swift-ci smoke test

**Comment by varungandhi-apple** [comment]
Potential addition: Should this be checking if `sccache`'s server is not on, and turn it on if that's not the case?

**Comment by porglezomp** [comment]
According to the sccache README, that shouldn't be necessary:
> The sccache command will spawn a server process if one is not already running

**Comment by varungandhi-apple** [comment]
Nice, I didn't realize that. We could get rid of the `--start-server` in the Getting Started guide then.

**Comment by edymtt** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
@swift-ci python lint

**Comment by shahmishal** [comment]
```
00:03:20.173 ./utils/swift_build_support/tests/test_cmake.py:135:13: E128 continuation line under-indented for visual indent
```

**Comment by edymtt** [comment]
@swift-ci please smoke test

**Comment by edymtt** [comment]
@swift-ci python lint

**Comment by edymtt** [comment]
@swift-ci please smoke test

**Comment by edymtt** [comment]
@swift-ci please test Windows

**Comment by shahmishal** [comment]
@swift-ci test windows

**Pull request merged**